core/capabilities/remote/target/caller.go

```diff
@@ -0,0 +1,142 @@
+package target
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/target/request"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// caller is a shim for remote target capabilities.
+// It translates between capability API calls and network messages.
+// Its responsibilities are:
+//  1. Transmit capability requests to remote nodes according to a transmission schedule
+//  2. Aggregate responses from remote nodes and return the aggregated response
+//
+// caller communicates with corresponding receiver on remote nodes.
+type caller struct {
+	lggr                 logger.Logger
+	remoteCapabilityInfo commoncap.CapabilityInfo
+	localDONInfo         capabilities.DON
+	dispatcher           types.Dispatcher
+	requestTimeout       time.Duration
+
+	messageIDToCallerRequest map[string]*request.CallerRequest
```

Comment:
Why did you choose to put the Request objects in separate packages? In my mind organising all of the objects that talk to one another into the same package feels like it makes more sense (so caller + caller request; receiver + its request).

Reply:
to prevent accidental call of non-threadsafe methods, just caution really